### PR TITLE
Added populate_board method to Game model

### PR DESCRIPTION
**Pull request by jonathanpike**
Also added a test!

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -1,4 +1,52 @@
...
+    init_pawn
+    init_rook
+    init_knight
+    init_bishop
+    init_queen
+    init_king
+  end
+
+  def init_pawn
+    7.downto(0) do |col|
```
Comment (conslo) [inline app/models/game.rb]: I like the loop, makes it much cleaner

File: app/models/game.rb [hunk 2]
```diff
@@ -1,4 +1,52 @@
...
+
+  def populate_board!
+    init_pawn
+    init_rook
+    init_knight
+    init_bishop
+    init_queen
+    init_king
+  end
+
```
Comment (conslo) [inline app/models/game.rb]: maybe make the separate piece init methods private? prevent people from trying to run them themselves. Let me know your thoughts

File: test/models/game_test.rb [hunk 3]
```diff
@@ -1,7 +1,33 @@
...
 
 class GameTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  def setup
+    @g = Game.create(name: "New Game", white_player_id: 1, black_player_id: 1)
+  end
+
+  test "board populated with pieces" do
```
Comment (conslo) [inline test/models/game_test.rb]: rename this test to more represent what it's testing, in this case that the count of pieces is correct

File: test/models/game_test.rb [hunk 4]
```diff
@@ -1,7 +1,33 @@
 require 'test_helper'
 
 class GameTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  def setup
+    @g = Game.create(name: "New Game", white_player_id: 1, black_player_id: 1)
```
Comment (conslo) [inline test/models/game_test.rb]: the white and black player should be different. Might not make a difference now, but the tests could be subtly wrong if you don't represent this

Comment (conslo) [inline test/models/game_test.rb]: Also, use factories to create the players in question.

File: app/models/game.rb [hunk 5]
```diff
@@ -1,4 +1,52 @@
...
+  def init_queen
+    pieces.create(type: "Queen", col_position: 3, row_position: 0)
+    pieces.create(type: "Queen", col_position: 3, row_position: 7)
+  end
+
+  def init_king
+    pieces.create(type: "King", col_position: 4, row_position: 0)
+    pieces.create(type: "King", col_position: 4, row_position: 7)
+  end
```
Comment (conslo) [inline app/models/game.rb]: None of these are specifying the player! D: So who are they owned by?

File: test/models/game_test.rb [hunk 6]
```diff
@@ -1,7 +1,33 @@
...
+  end
+
+  test "board populated with pieces" do
+    expected = 32
+    actual = @g.pieces.count
+    assert_equal expected, actual
+  end
+
```
Comment (conslo) [inline test/models/game_test.rb]: Add two tests, similar to the one above, that tests for the correct count of pieces for both black and white

File: app/models/game.rb (hunk 2 above)
Comment (jonathanpike) [inline app/models/game.rb]: We initially had the init methods private, but the way we wrote them made it so they weren't associated with a game.  We re-wrote them given Rubocops suggestion, and never tried making them private again.  I tested it and making them private works. Thanks for the suggestion.

File: test/models/game_test.rb (hunk 3 above)
Comment (jonathanpike) [inline test/models/game_test.rb]: Renamed.

File: test/models/game_test.rb (hunk 4 above)
Comment (jonathanpike) [inline test/models/game_test.rb]: Thanks -- I was following Ken's example in the gist (ie. when he was testing via rails console), and didn't think about how that would change things.  I've changed it.

File: app/models/game.rb (hunk 5 above)
Comment (jonathanpike) [inline app/models/game.rb]: Dumb mistake.  Sorry about that -- fixed!

File: test/models/game_test.rb (hunk 6 above)
Comment (jonathanpike) [inline test/models/game_test.rb]: Added the tests.

**Comment by conslo** [comment]
lgtm!

**Pull request merged**